### ISSUE TITLE
NPE when SELECT with aggregation by partitioned table

### ISSUE THREAD

**Issue by shm-dmitry** [crate/crate#9725]
**CrateDB version**: 4.0.11

**Problem description**:

I have table partitioned by nested column.
I receive NPE when SELECT from this table with GROUP BY by partition key.

**Steps to reproduce**:

```
create table test (
    pk object as (
        id string, 
        part string
    ), 
   PRIMARY KEY ("pk"['id'], "pk"['part'])
) 
partitioned by (pk['part']);
```
`>> Table TEST created.`

`insert into test (pk) values ({ id = '1', part = 'x' });`
`>> 1 row inserted.`

`select distinct pk['part'] from test;`
**`>> ERROR: UnhandledServerException: java.lang.NullPointerException`**

`select  pk['part'], count(*) from test group by pk['part'];`
**`>> ERROR: UnhandledServerException: java.lang.NullPointerException`**

**Comment by mfussenegger** [comment]
Thank you for the report. Looks like this is already fixed in master but still broken in 4.1.

We will see which change fixed it and backport it to 4.1

**Comment by mfussenegger** [comment]
We merged a fix (https://github.com/crate/crate/pull/9730) which will be released with 4.1.3
